File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>

using namespace std;
bool checkSkip(char c, string skip){
    for(int i=0;i<skip.length();i++){
        if(c==skip[i]) return false;
    }
    return true;
}
string solution(string s, string skip, int index) {
    string answer = "";
    for(int i=0;i<s.length();i++){
        char c=s[i];
        for(int j=0;j<index;){
            c++;
            int tmp=int(c);
            if(tmp>122){
                c=static_cast<char>(tmp-123+97);
            }
            if(checkSkip(c, skip)==true){
                j++;
            }
        }
        answer+=c;
        
    }
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;
class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        for(int i=0;i<s.length();i++){
            int start=s.charAt(i);
            int tmp=0;
            for(int j=0;j<index;j++){
                start++;
                if(start>122) start-=26;
                int k=0; 
                while (k<skip.length()) { 
                    if(skip.charAt(k)==start){
                        start++;
                        if(start>122) start-=26;
                        k=0;
                    } else {
                        k++; 
                    }
                }
            }
            if(start>122) start-=26;
            answer+=(char)start;
        }
        return answer;
    }
}